distributed/protocol/serialize.py

```diff
@@ -541,6 +534,21 @@ def deserialize_bytes(b):
     return deserialize(header, frames)
 
 
+def serialize_bytes(x, **kwargs):
+    L = serialize_bytelist(x, **kwargs)
+    return b"".join([pack_frames_prelude(L), *L])
+
+
+def deserialize_bytes(b):
+    try:
+        b = memoryview(b)
+    except TypeError:
```

Comment:
I'm sorry, I think I've asked this before, can you remind me when you think we would hit a `TypeError`?

Reply:
If the object is not [`bytes`-like]( https://docs.python.org/3/glossary.html#term-bytes-like-object ), this will raise. We end up performing this coercion in [`unpack_frames`]( https://github.com/dask/distributed/blob/a848ee0c20d4b3df261da562c4c66cb762d2a3ec/distributed/protocol/utils.py#L116 ) later. So this merely performs this coercion a bit sooner.

The non-exception case is for things like spilling to disk (analogous to `deserialize_bytes(serialize_bytes(obj))`) and the exception case is for things like compressed in-memory spilling (analogous to `deserialize_bytes(serialize_bytelist(obj))`). IOW both cases can rely on using `deserialize_bytes`.

Reply:
Thank you!